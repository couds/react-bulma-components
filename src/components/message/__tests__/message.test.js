import PropTypes from 'prop-types';
import React from 'react';
import { render } from 'react-testing-library';
import { Message } from '..';

describe('Message component', () => {
  it.each([[Message], [Message.Header], [Message.Body]])('should render', Component => {
    const { asFragment } = render(<Component>Content</Component>);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render with all', () => {
    const { asFragment } = render(
      <Message>
        <Message.Header>Lorem Ipsum</Message.Header>
        <Message.Body>Lorem Ipsum</Message.Body>
      </Message>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should concat classname in props with classname', () => {
    const { asFragment } = render(
      <Message className="other-class">
        <Message.Header>Lorem Ipsum</Message.Header>
        <Message.Body>Lorem Ipsum</Message.Body>
      </Message>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render as an html section', () => {
    const { asFragment } = render(<Message renderAs="section">This should be a section</Message>);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should have custom inline styles', () => {
    const { asFragment } = render(
      <Message renderAs="section" style={{ width: 200, zIndex: 1 }}>
        This should be a section with custom styles
      </Message>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should accept a react Element as renderAs prop', () => {
    const Custom = props => (
      <p {...props}>
        Custom
        {props.children}
      </p>
    );
    Custom.propTypes = { children: PropTypes.node.isRequired };

    const { asFragment } = render(<Message renderAs={Custom}>This should be a p element</Message>);
    expect(asFragment()).toMatchSnapshot();
  });
});
