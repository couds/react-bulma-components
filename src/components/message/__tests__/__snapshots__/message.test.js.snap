// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component should accept a react Element as renderAs prop 1`] = `
<DocumentFragment>
  <p
    class="message"
  >
    CustomThis should be a p element
  </p>
</DocumentFragment>
`;

exports[`Message component should concat classname in props with classname 1`] = `
<DocumentFragment>
  <article
    class="message other-class"
  >
    <div
      class="message-header"
    >
      Lorem Ipsum
    </div>
    <div
      class="message-body"
    >
      Lorem Ipsum
    </div>
  </article>
</DocumentFragment>
`;

exports[`Message component should have custom inline styles 1`] = `
<DocumentFragment>
  <section
    class="message"
    style="width: 200px; z-index: 1;"
  >
    This should be a section with custom styles
  </section>
</DocumentFragment>
`;

exports[`Message component should render 1`] = `
<DocumentFragment>
  <article
    class="message"
  >
    Content
  </article>
</DocumentFragment>
`;

exports[`Message component should render 2`] = `
<DocumentFragment>
  <div
    class="message-header"
  >
    Content
  </div>
</DocumentFragment>
`;

exports[`Message component should render 3`] = `
<DocumentFragment>
  <div
    class="message-body"
  >
    Content
  </div>
</DocumentFragment>
`;

exports[`Message component should render as an html section 1`] = `
<DocumentFragment>
  <section
    class="message"
  >
    This should be a section
  </section>
</DocumentFragment>
`;

exports[`Message component should render with all 1`] = `
<DocumentFragment>
  <article
    class="message"
  >
    <div
      class="message-header"
    >
      Lorem Ipsum
    </div>
    <div
      class="message-body"
    >
      Lorem Ipsum
    </div>
  </article>
</DocumentFragment>
`;
